### PR TITLE
feature(readme): add an application overview to the README.md

### PR DESCRIPTION
**Pull request by Jonas-Karl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
...
+### Application Layer
+- parsing
+  - `EmailParser`: An interface to parse Email of any format to a domain Email object.
+  - `EmailParsingException`: A custom exception to handle parsing errors.
+
+
+### Infrastructure Layer
+- emails:
+  - `EmailMetadataFactory`: A factory to create domain EmailMetadata objects by passing information of the external library jakarta.
+  - `EmlParser`: An implementation of the `EmailParser` interface to parse .eml files to domain Email objects.
```
Comment (jens1o) [inline README.md:56]: reminder for myself to add my ImapListener stuff to here :)

but looks pretty decent otherwise

**Review by jens1o — APPROVED** [review]
good job! I couldn't come up with things that went missing, it seems quite thorough to me. Excited to see what Lars comes up with :)

File: README.md [hunk 2]
```diff
@@ -1 +1,56 @@
-# mailserver
\ No newline at end of file
+# mailserver
+
+This server is supposed to handle TCP connections to receive emails from clients.
```
Comment (seiferla) [inline README.md:3]: Did we not say that we can also send a e-mail with smpt to another server or was this an optional use case?

This server is supposed to handle TCP connections to receive emails with IMAP and send emails with SMTP to another server.

File: README.md [hunk 3]
```diff
@@ -1 +1,56 @@
...
+    - [ ] Create a Mailbox with multiple Emails.
+  - [ ] Store Emails in a Folder.
+  - [ ] Send Emails to the cc,bcc, to-Addresses.
+- [ ] Synchronization
+  - [ ] provide a list of emails when a client connects.
+  - [ ] provide a list of emails when a client requests a refresh.
+  - [ ] provide a list of emails with according folders instead.
+- [ ] Filtering
+  - [ ] Filter incoming messages Spam to a spam folder
+
```
Comment (seiferla) [inline README.md:20]: Should we also add an point - [ ] Authorization/Authentification ?

File: README.md [hunk 4]
```diff
@@ -1 +1,56 @@
...
+## Domains
+
+## Ubiquitous Language
+- `Recieving Email` : Emails that are sent to this server. This can happen by a client or in the future by another mailserver
```
Comment (seiferla) [inline README.md:24]: Oh, okay. Now the definition above is now clear for me.  But maybe the definition is a bit too late or we change the definition of our mail server and describe it a bit more generally. line 3-4

File: README.md (hunk 2 above)
Comment (Jonas-Karl) [inline README.md:3]:
```suggestion
This server is supposed to handle TCP connections to receive emails from clients. A connection will be possible via IMAP as well as SMTP.
```

File: README.md (hunk 3 above)
Comment (Jonas-Karl) [inline README.md:20]:
```suggestion
- [ ] Authorization / Authentication
  - [x] implicit registration
  - [ ] login to get access to own mailbox

```

File: README.md (hunk 4 above)
Comment (Jonas-Karl) [reply README.md:24]: I added IMAP and SMTP to the server definition

**Review by Jonas-Karl — COMMENTED** [review]
add suggestions of @seiferla

**Comment by Jonas-Karl** [comment]
You can have another look at it

File: README.md [hunk 5]
```diff
@@ -1 +1,59 @@
...
\ No newline at end of file
+# mailserver
+
+This server is supposed to handle TCP connections to receive emails from clients. A connection will be possible via IMAP as well as SMTP.
+The vision is a fully functional mailserver to manage your emails.
+
+
+## Functionality
+- [ ] Receive Emails
+  - [x] Parse incoming Emails .
```
Comment (jens1o) [inline README.md:9]:
```suggestion
  - [x] Parse incoming Emails.
```

File: README.md [hunk 6]
```diff
@@ -1 +1,59 @@
...
+The vision is a fully functional mailserver to manage your emails.
+
+
+## Functionality
+- [ ] Receive Emails
+  - [x] Parse incoming Emails.
+  - [ ] Store Emails in a Mailbox.
+    - [ ] Create a Mailbox with multiple Emails.
+  - [ ] Store Emails in a Folder.
+  - [ ] Send Emails to the cc,bcc, to-Addresses.
```
Comment (jens1o) [inline README.md:13]:
```suggestion
  - [ ] Send Emails to the cc, bcc, to-Addresses.
```

**Pull request merged**